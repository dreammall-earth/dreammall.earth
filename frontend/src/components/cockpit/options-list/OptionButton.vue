<template>
  <li class="item" :class="{ red }">
    <button class="button" @click="$emit('click')">
      <slot />
    </button>
  </li>
</template>

<script lang="ts" setup>
defineProps<{
  red?: boolean
}>()
defineEmits(['click'])
</script>

<style scoped>
.item {
  width: calc(54px + var(--padding-left, 0px) + var(--padding-right, 0px));
  padding-right: var(--padding-right, initial);
  padding-left: var(--padding-left, initial);
  font-size: 10px;
  background: var(--v-cockpit-element-background);

  &:first-child {
    border-radius: 16px 0 0 16px;

    --padding-left: 4px;
  }

  &:last-child {
    border-radius: 0 16px 16px 0;

    --padding-right: 4px;
  }
}

.item.red {
  color: #f5f5f5;
  background-color: #d02f44;
}

.button {
  display: flex;
  flex-flow: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}
</style>
