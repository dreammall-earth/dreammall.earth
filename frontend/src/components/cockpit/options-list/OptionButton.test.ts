import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

import OptionButton from './OptionButton.vue'

describe('Option Button', () => {
  const Wrapper = () => mount(OptionButton)

  it('renders', () => {
    const wrapper = Wrapper()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('emits click event when clicked', async () => {
    const wrapper = Wrapper()
    await wrapper.find('button').trigger('click')
    expect(wrapper.emitted('click')).toBeTruthy()
  })
})
