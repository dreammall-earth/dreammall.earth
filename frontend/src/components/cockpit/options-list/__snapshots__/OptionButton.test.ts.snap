// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Option Button > renders 1`] = `
<li
  class="item"
  data-v-b0dd0a64=""
>
  <button
    class="button"
    data-v-b0dd0a64=""
  >
    
    
  </button>
</li>
`;
